src/app/cards/detail/[id]/page.tsx

```diff
@@ -0,0 +1,86 @@
+import { revalidatePath } from 'next/cache';
+import Link from 'next/link';
+import { notFound } from 'next/navigation';
+import React from 'react';
+
+import { DeleteButton } from './DeleteButton';
+
+import { TopNav } from '@/components/TopNav';
+import { BackIcon } from '@/components/icons/BackIcon';
+import { ShareIcon } from '@/components/icons/ShareIcon';
+import { deleteCard, getCard } from '@/services/cardService';
+import { getType } from '@/services/typeService';
+
+export default async function Page({ params }: { params: { id: string } }) {
+    const { card } = await getCard(params.id);
+    if (!card) {
+        notFound();
+    }
+
+    const backBtn = (
+        <Link href={`/cards/${params.id}`}>
+            <BackIcon />
+        </Link>
+    );
+
+    const btnPanel = (
+        <div className="flex gap-2">
+            <Link href="/">
+                {/* TODO: Implement share card function  */}
+                <ShareIcon />
+            </Link>
+        </div>
+    );
+
+    const { type } = await getType(card.type);
```

Comment:
We can use populate function to get type data in card service. Created a [ticket](https://tsoibet.atlassian.net/browse/KAADO-40) to address this.